utils/build_stig_control.py

```diff
@@ -121,41 +121,88 @@ def check_files(args):
         exit(-1)
 
 
-def get_controls(known_rules, ns, root) -> list:
+def get_extra_srgs(rule, ns) -> list:
+    pattern = re.compile(r'SRG-[A-Z]{2,}-\d{5,}-[A-Z]{3,}-\d{5,}')
+
+    description = rule.find('checklist:description', ns).text
+    return pattern.findall(description)
+
+
+def get_rules_for_control(stig_id, known_rules, srgs, srg_controls):
+    # Add any known rule with the same STIG ID reference
+    rule_set = set()
+    if stig_id in known_rules.keys():
+        rule_set.update(known_rules.get(stig_id))
+
+    # Let's also add any rule selected in the SRG control file
+    if srg_controls:
+        for srg in srgs:
+            rule_set.update(srg_controls.get_control(srg).rules)
+
+    return sorted(list(rule_set))
+
+
+def get_controls(known_rules, ns, root, srg_controls=None) -> list:
     controls = list()
     for group in root.findall('checklist:Group', ns):
+        # There is always at least one SRG associated
+        srgs = [group.find('checklist:title', ns).text]
+
         for stig in group.findall('checklist:Rule', ns):
             stig_id = stig.find('checklist:version', ns).text
+            # Add any other SRG associated mentioned in description
+            srgs += get_extra_srgs(stig, ns)
             control = dict()
             control['id'] = stig_id
             control['levels'] = [stig.attrib['severity']]
             control['title'] = stig.find('checklist:title', ns).text
-            if stig_id in known_rules.keys():
-                control['rules'] = known_rules.get(stig_id)
+            control['rules'] = get_rules_for_control(stig_id, known_rules, srgs, srg_controls)
+            if len(control['rules']) > 0:
                 control['status'] = 'automated'
```

Comment:
I wonder if we also need to check if the rule has any ovals/template, some rules could just be a manual rule.

Reply:
That is a good question.
The original script didn't check for that either, this could be done in subsequent PRs.

Reply:
Not directly related to this comment, but I have gone through the controls that were  `pending` and updated their status.